tmd/view/plot.py

```diff
@@ -34,10 +34,13 @@ def barcode(ph, new_fig=True, subplot=False, color="b", linewidth=1.2, **kwargs)
     ph_sort = analysis.sort_ph(ph)
 
     for ip, p in enumerate(ph_sort):
-        ax.plot(p[:2], [ip, ip], c=color, linewidth=linewidth)
+        if not isinstance(color, list):
```

Comment:
You could create the color variable using the check and then make the call once.

```python
bar_color = color[ip] if isinstance(color, list) else color
```